src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -60,171 +38,25 @@ std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_
         if (std::count(robots_to_ignore.begin(), robots_to_ignore.end(),
                        friendly_robot) == 0)
         {
-            obstacles.emplace_back(Circle(friendly_robot.position(), radius));
+            obstacles.emplace_back(friendly_robot);
         }
     }
 
     std::optional<Shot> best_shot;
 
     // Calculate the best_shot based on what goal we're shooting at
-    if (shoot_on_enemy_goal)
+    if (goal == TeamType::FRIENDLY)
     {
-        best_shot = calcBestShotOnGoal(field.enemyGoalpostNeg(), field.enemyGoalpostPos(),
-                                       point, obstacles);
+        best_shot = calcBestShotOnGoal(
+            Segment(field.friendlyGoalpostNeg(), field.friendlyGoalpostPos()),
+            shot_origin, obstacles);
```

Comment:
nit: 
```suggestion
       return calcBestShotOnGoal(
            Segment(field.friendlyGoalpostNeg(), field.friendlyGoalpostPos()),
            shot_origin, obstacles);
```
Do the same below and then we don't need `best_shot`

Reply:
Done.